src/main/java/com/techweek/n8n_demo/controller/UserController.java

```
@@ -0,0 +1,58 @@
+package com.techweek.n8n_demo.controller;
+
+import com.techweek.n8n_demo.entity.User;
+import com.techweek.n8n_demo.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping
+    public List<User> listUsers() {
+        return userService.getAllUsers();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<User> getUser(@PathVariable Long id) {
+        return userService.getUserById(id)
+                .map(ResponseEntity::ok)
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+    @GetMapping("/by-email")
+    public ResponseEntity<User> getByEmail(@RequestParam String email) {
+        Optional<User> user = userService.getUserByEmail(email);
+        return user.map(ResponseEntity::ok).orElse(ResponseEntity.notFound().build());
+    }
+
+    @PostMapping
+    public ResponseEntity<User> create(@RequestBody User user) {
+        if (user.getEmail() == null || userService.emailExists(user.getEmail())) {
+            return ResponseEntity.badRequest().build();
+        }
+        User created = userService.createUser(user);
+        return ResponseEntity.created(URI.create("/api/users/" + created.getId())).body(created);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<User> update(@PathVariable Long id, @RequestBody User user) {
+        Optional<User> updated = userService.updateUser(id, user);
+        return updated.map(ResponseEntity::ok).orElse(ResponseEntity.notFound().build());
+    }
+
+    @DeleteMapping("/{id}")
```

The `deleteUser` method in `UserService` is called without checking if the user actually exists. This might lead to silent failures if an invalid ID is provided. The `delete` method in the controller returns `204 No Content`, which implies a successful deletion, even if the user didn't exist.